src/NzbDrone.Core/MediaFiles/DiskScanService.cs

```diff
@@ -89,6 +89,19 @@ public void Scan(List<string> folders = null, FilterFilesType filter = FilterFil
             {
                 authorIds = new List<int>();
             }
+            else
+            {
+                folders = new List<string>();
+                foreach (var authorId in authorIds)
+                {
+                    var author = _authorService.GetAuthor(authorId);
+
+                    if (author != null)
+                    {
+                        folders.Add(author.Path);
```

Comment:
This is one part I have an issue with - it assumes that you only need to scan the author's folder

Reply:
> This is one part I have an issue with - it assumes that you only need to scan the author's folder

I'm confused, doesn't the function have an AuthorIDs array variable that's only passed in if we should only be looking at the authors it contains? not re'evaluating the root folder. Right?